my-app/src/CreateCategory.js

```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import CreateCategoryInfo from './CreateCategoryInfo.js';
+import CreateItemDescription from './CreateItemDescription.js';
+import categories from './categories.json';
+import menuItems from './menuItems.json';
+
+export default function CreateCategory(props) {
+  function addItemByCategoryInMap(map, key, menuItem) {
```

Comment:
This function will unnecessarily be recreated each time this component will get rendered. 

Move such helper functions outside to a `helpers` folder inside the component folder.